hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperation.java

```diff
@@ -464,4 +264,183 @@ protected void writeInternal(ObjectDataOutput out) throws IOException {
     public int getId() {
         return MapDataSerializerHook.ENTRY_OPERATION;
     }
+
+    private final class OffloadImpl extends Offload {
+        private OffloadImpl() {
+            super(EntryOperation.this);
+        }
+
+        @Override
+        public void start() {
+            verifyEntryProcessor();
+
```

Comment:
This should be done on init to avoid the offloading cycle in case of validation issues.
Maybe it would be good to add a abstract validate() method to the Offload class